Library/Source/NativeEngine.cpp

```diff
@@ -277,6 +320,155 @@ namespace babylon
             bgfx::TextureHandle Texture{ bgfx::kInvalidHandle };
         };
 
+        class ViewClearState
+        {
+        public:
+            ViewClearState(uint16_t viewId)
+                : m_viewId{ viewId }
+            {}
+
+            bool Update(const Napi::CallbackInfo& info)
+            {
+                auto r = info[0].As<Napi::Number>().FloatValue();
+                auto g = info[1].As<Napi::Number>().FloatValue();
+                auto b = info[2].As<Napi::Number>().FloatValue();
+                auto a = info[3].IsUndefined() ? 1.f : info[3].As<Napi::Number>().FloatValue();
+                auto backBuffer = info[4].IsUndefined() ? true : info[4].As<Napi::Boolean>().Value();
+                auto depth = info[5].IsUndefined() ? true : info[5].As<Napi::Boolean>().Value();
+                auto stencil = info[6].IsUndefined() ? true : info[6].As<Napi::Boolean>().Value();
+
+                bool needToUpdate = r != m_red
+                    || g != m_green
+                    || b != m_blue
+                    || a != m_alpha
+                    || backBuffer != m_backBuffer
+                    || depth != m_depth
+                    || stencil != m_stencil;
+                if (needToUpdate)
+                {
+                    m_red = r;
+                    m_green = g;
+                    m_blue = b;
+                    m_alpha = a;
+                    m_backBuffer = backBuffer;
+                    m_depth = depth;
+                    m_stencil = stencil;
+
+                    Update();
+                }
+
+                return needToUpdate;
+            }
+
+            void Update() const
+            {
+                // TODO: Backbuffer, depth, and stencil.
+                bgfx::setViewClear(m_viewId, BGFX_CLEAR_COLOR | (m_depth ? BGFX_CLEAR_DEPTH : 0x0), Color());
```

Comment:
Note: With opengl/webgl, if you bind a framebuffer and call clear, the buffer is cleared. With bgfx, I think the view is cleared if there is at least 1 drawcall. You may have to call bgfx::touch for a dummy DC.

Reply:
Okay. I don't currently have a test case that shows any issues related to that, at least as far as I know. I'd rather tackle that when I have a usage that shows the problem, so that I can know what kind of problem it is and that I've actually fixed it. Do you have a test case that exhibits this such that we can tackle it now, or should we cross that bridge when we come to it?

Reply:
in your js sample, create the BABYLON.RenderTargetTexture, do the call to clear the view (let's say with purple color), don't render any mesh into it. Use the rendertargettexture as diffuse on a plane. The plane should be purple in your browser/playbground. but I think the view will not be cleared with bgfx without the touch and the plane will be black.

Reply:
it's a minor issue and should not prevent to merge and move forward :)

Reply:
Good call, worked exactly as you said it would, and adding bgfx::touch() cleared it right up, fix confirmed. Thanks!